service-sdk/16.0.0/service-to-service-communication/example-service-to-service/src/main/java/com/backbase/example/service/GreetingsServiceImpl.java

```diff
@@ -19,28 +20,33 @@ public class GreetingsServiceImpl implements GreetingsService {
     private MessageApi exampleIntegrationApiClient;
 
     @Override
+    @Transactional(readOnly = true)
     public List<Greeting> getGreetings() {
         return greetingsRepository.findAll();
     }
 
     // tag::getExternalGreetings[]
     @Override
+    @Transactional(readOnly = true)
     public List<Greeting> getExternalGreetings() {
         return GreetingsMapper.INSTANCE.integrationMessageToGreeting(exampleIntegrationApiClient.getMessages());
     }
     // end::getExternalGreetings[]
 
     @Override
+    @Transactional(readOnly = true)
     public Greeting getGreetingById(String id) {
         return greetingsRepository.findById(id).get();
     }
 
     @Override
+    @Transactional
```

Comment:
we can not add `@Transactional` here because it is by default.

Reply:
Removed!